### PR TITLE
move js demo examples to ts

### LINKED ISSUE #2772: Merge js demo into ts demo
Currently we maintain three demo projects, react-native, ts, and js. As we are deprecating prop-types and migrating fully to TS we can remove the js demo project.

We should go through all demos in the JS project and ensure we have a matching demo in the TS project so we don't lose any use cases.

### PR DESCRIPTION
**Pull request by Burnett2k**
### Description

We are working to remove the code within the `demo/js` folder, so that we don't have to maintain examples in the javascript AND typescript demo. Before deleting, I went through and compared the demos to see if anything was missing.

Findings:
* a few typescript issues that needed to be fixed to reduce the # of ts errors in the demo project
* mostly needed to copy / paste over examples that didn't get ported over for some reason

Fixes #2772 

#### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### How Has This Been Tested?

I went through the ts demo project to ensure examples I modified are showing up correctly.

### Checklist: (Feel free to delete this section upon completion)

- [x] I have included a changeset if this change will require a version change to one of the packages.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have run all builds, tests, and linting and all checks pass
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/victory-candlestick/src/victory-candlestick.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export interface VictoryCandlestickStyleInterface {
 export type VictoryCandlestickLabelsType =
   | (string | number)[]
   | boolean
-  | ((datum: any) => number);
+  | ((datum: any) => number | string);
```
Comment (Burnett2k) [inline packages/victory-candlestick/src/victory-candlestick.tsx:43]: @KenanYusuf Mind double checking this fix I put in? There was a TS error in the candlestick demo caused by this type not allowing the anon function to return string type. I assume that should be allowed as that's how it was being used in the demo.

Comment (KenanYusuf) [reply packages/victory-candlestick/src/victory-candlestick.tsx:43]: Yeah, this seems correct, good catch.

File: demo/ts/components/create-container-demo.tsx [hunk 2]
```diff
@@ -277,6 +273,7 @@ class App extends React.Component {
   render() {
     return (
       <div className="demo">
+        hi
```
Comment (KenanYusuf) [inline demo/ts/components/create-container-demo.tsx:276]:
```suggestion
```
Accidental hi? 😄

File: demo/ts/components/victory-tooltip-demo.tsx [hunk 3]
```diff
@@ -86,6 +68,35 @@ class App extends React.Component {
             ]}
           />
 
+          <VictoryCandlestick
+            horizontal
+            style={{ parent: parentStyle }}
+            highLabelComponent={<VictoryTooltip active dy={-3} />}
+            // @todo high labels type needs to have an OR so that it can also return a string.
+            // Currently, it only allows returning an integer which is incorrect, as a label would typically be a string.
```
Comment (KenanYusuf) [inline demo/ts/components/victory-tooltip-demo.tsx:76]: Can we remove this comment now you've updated the type?

File: demo/ts/components/create-container-demo.tsx (hunk 2 above)
Comment (Burnett2k) [reply demo/ts/components/create-container-demo.tsx:276]: wow, good catch 🙃

File: demo/ts/components/victory-tooltip-demo.tsx (hunk 3 above)
Comment (Burnett2k) [reply demo/ts/components/victory-tooltip-demo.tsx:76]: for sure

File: packages/victory-candlestick/src/victory-candlestick.tsx (hunk 1 above)
Comment (Burnett2k) [reply packages/victory-candlestick/src/victory-candlestick.tsx:43]: Kewl, thanks for checking!

File: package.json [hunk 4]
```diff
@@ -128,8 +128,7 @@
     "version": "pnpm changeset version && pnpm install --no-frozen-lockfile",
     "publish": "nps git:tag && pnpm changeset publish --no-git-tag",
     "changeset": "changeset",
-    "start": "concurrently --raw \"pnpm:build:lib:esm --watch\" \"webpack serve --config ./config/webpack/demo/webpack.config.dev.js --static demo/js --entry ./demo/js/app\"",
-    "start:ts": "concurrently --raw \"pnpm:build:typescript --watch\" \"webpack serve --config ./config/webpack/demo/webpack.config.dev.js --static demo/ts --entry ./demo/ts/app\"",
+    "start": "concurrently --raw \"pnpm:build:typescript --watch\" \"webpack serve --config ./config/webpack/demo/webpack.config.dev.js --static demo/ts --entry ./demo/ts/app\"",
```
[the remark this answers is not in the document]
Comment (KenanYusuf) [reply package.json:131]: `start` makes sense now there's no js demos 👍

**Review by KenanYusuf — APPROVED** [review]

**Pull request merged**